Feature/movie details booking page

krushna-simform opened the pull request:
# Description

- Implemented `MovieDetails` and `MoviteBooking` Page.
- Added Search functionality to Application.
- Create a separate component if components are repetitive.

# Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Refractoring
- [ ] Performance improvement
- [ ] Scaffolding

# Screenshots

![Screenshot from 2025-05-08 18-04-20](https://github.com/user-attachments/assets/478b33cc-90d1-4da0-bb6d-eab941c0ca11)
![Screenshot from 2025-05-08 18-04-34](https://github.com/user-attachments/assets/a51aa592-9857-47ae-a81a-14880f3da168)


# Validations

![Screenshot from 2025-05-09 10-52-36](https://github.com/user-attachments/assets/9b292820-43e9-4414-bbfb-72122e96b5aa)
![Screenshot from 2025-05-09 10-53-18](https://github.com/user-attachments/assets/eeaa0b0d-87e8-4b7b-9ac9-5d44c19dafba)
![Screenshot from 2025-05-09 10-50-23](https://github.com/user-attachments/assets/7a95fae1-5276-4eec-b1ab-040b598ea85c)


# How to test

Deploy Preview: https://filmfusion-git-feature-ef3f82-krushna-rathods-projects-01be27f9.vercel.app/

krushna-simform committed:
RESTRUCTURE: move movieData.ts into data folder

krushna-simform committed:
RESTRUCTURE: move MainFeedPage.tsx into pages folder

krushna-simform committed:
FEATURE: update imports based on file restructure

krushna-simform committed:
FEATURE: update button ui and Header component

krushna-simform committed:
FEATURE: create MovieDetailsPage

krushna-simform committed:
FEATURE: add react router dom

krushna-simform committed:
FEATURE: update interface for movie details

krushna-simform committed:
FEATURE: based on interface update movie data

krushna-simform committed:
FEATURE: base on interface update MovieCard and MainFeed

krushna-simform committed:
FEATURE: align haeder and mainfeed in center

krushna-simform committed:
FEATURE: add navigation in MovieCard

krushna-simform committed:
FEATURE: create Image.tsx component

krushna-simform committed:
FEATURE: update movie data

krushna-simform committed:
FEATURE: create Ratings component

krushna-simform committed:
FEATURE: create AboutMovie component

krushna-simform committed:
FEATURE: create Cast component

krushna-simform committed:
FEATURE: create Director component

krushna-simform committed:
FEATURE: create DetailsPageMovieCard

krushna-simform committed:
FEATURE: create MovieDetailsShocse component

krushna-simform committed:
FEATURE: create MovieDetailsDescription component

krushna-simform committed:
FEATURE: update MovieDetailsPage

krushna-simform committed:
FEATURE: add required icons

krushna-simform committed:
FEATURE: create movie timing component

krushna-simform committed:
FEATURE: create MovieBookingSeats component

krushna-simform committed:
FEATURE: create MovieBooking component

krushna-simform committed:
FEATURE: create MovieBookingPage component

krushna-simform committed:
FEATURE: add navigation in Showcase section

krushna-simform committed:
FEATURE: add routing path for booking page

krushna-simform committed:
FEATURE: add search functionality

krushna-simform committed:
FEATURE: delete unwanted file

krushna-simform committed:
FEATURE: update routes

krushna-simform committed:
FEATURE: delete bg image

krushna-simform committed:
FEATURE: add bg gradient

krushna-simform committed:
FEATURE: update movie data

krushna-simform committed:
FEATURE: add vercel.json file

krushna-simform committed:
FEATURE: add aria-label in header

krushna-simform committed:
FEATURE: update header element

dhruv-d-simform commented:
![Screenshot from 2025-05-09 11-54-46](https://github.com/user-attachments/assets/830ab993-61bb-4f16-8882-6f430d95e310)

Can we add a link to home page on this "FilmFusion" logo ?

And can we also add cursor pointer to these movie timing buttons ?

dhruv-d-simform commented:
![Screenshot from 2025-05-09 11-57-48](https://github.com/user-attachments/assets/9f80090d-ee43-4dcd-93f1-f2ce28bf69ad)
Can we improve the UI for this ?

krushna-simform committed:
FEATURE: add link to header title

krushna-simform committed:
FEATURE: add cursor pointer for interactivity

krushna-simform committed:
FEATURE: create 404 error page

krushna-simform committed:
FEATURE: add 404 page not found component into pages

krushna-simform committed:
FEATURE: add 404 defult path in main routes

krushna-simform commented:
> ![Screenshot from 2025-05-09 11-54-46](https://private-user-images.githubusercontent.com/194202800/442004191-830ab993-61bb-4f16-8882-6f430d95e310.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY3NzM3NDEsIm5iZiI6MTc0Njc3MzQ0MSwicGF0aCI6Ii8xOTQyMDI4MDAvNDQyMDA0MTkxLTgzMGFiOTkzLTYxYmItNGYxNi04ODgyLTZmNDMwZDk1ZTMxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwOVQwNjUwNDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMzM1ZWZlMGEwM2QyNTUxZDlhNWE3ODA5YWFhMTJhNDNhMzRjYzQ4N2FiYzZjYzNkODZiYjkwYmI1OTFmMjhmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.d_sDKoXUaUyFiEy5mqFzz1-byvKswMs60t5sU1tYYvk)
> 
> Can we add a link to home page on this "FilmFusion" logo ?
> 
> And can we also add cursor pointer to these movie timing buttons ?

- I have added Link tag to `FilmFusion` Logo (https://github.com/krushna-simform/filmfusion/pull/3/commits/ee7ce449f1226f24b91b757f9b73882b35f067ef)

- Added cursor pointer (https://github.com/krushna-simform/filmfusion/pull/3/commits/202f950df0b185b59ed5446e5208a4396191e939)

krushna-simform commented:
> ![Screenshot from 2025-05-09 11-57-48](https://private-user-images.githubusercontent.com/194202800/442004884-9f80090d-ee43-4dcd-93f1-f2ce28bf69ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY3NzM3NDEsIm5iZiI6MTc0Njc3MzQ0MSwicGF0aCI6Ii8xOTQyMDI4MDAvNDQyMDA0ODg0LTlmODAwOTBkLWVlNDMtNGRjZC05M2YxLWYyY2UyOGJmNjlhZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwOVQwNjUwNDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMjU3M2M4MTZjMTFmODFmNTgxYTI2MjZkMzY3MWMyNDFkYzBlZGFmZGZkMTc3MDI4NGE3YmRiM2FkNGFmNGFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.o43YZZzLS1onZn46CiKqGL2gPlvulle9Q1tLfz4mNsQ) Can we improve the UI for this ?

I have created 404 error page (https://github.com/krushna-simform/filmfusion/pull/3/commits/47e119dfd1bde22deb4364c06ae18a10e8825101)

Kaushal-SF commented on src/utils/interface.ts:
Instead of `utils` folder, we can create a `types` folder to hold all our types.

krushna-simform replied:
I have Renamed utils folder to types (https://github.com/krushna-simform/filmfusion/pull/3/commits/a0acbe8ecdb77e397da86bfcf91b32df41e93cb4)

Kaushal-SF commented:
@krushna-simform Can we consider creating a atomic PRs instead of raising the large PR like this? I know there must be a mock data, other configuration and style changes available. However, this is not recommended to raise a single PR (with **1725** lines of changes).

<img width="238" alt="image" src="https://github.com/user-attachments/assets/7d903d14-966d-447f-92d5-896f92b1863e" />

Kaushal-SF commented on src/components/DetailsPageMovieCard.tsx:
```diff
@@ -0,0 +1,13 @@
+export const DetailsPageMovieCard = ({
+  title,
+  backPosterUrl,
+}: Pick<MovieDetails, "title" | "backPosterUrl">) => {
```
Nit: Good use case of `Pick`.

However, we can keep it outside the component. It will help if we want to add more types without ruining the readability.

krushna-simform replied:
I have moved props type outside of the componenet(https://github.com/krushna-simform/filmfusion/pull/3/commits/bb190f3ce4263e94094eef9036cc649515d72e4a)

Kaushal-SF commented on src/components/DetailsPageMovieCard.tsx:
Suggestion for file name:

- MovieDetailsCard.tsx
- MovieInfoCard.tsx
- MovieOverviewCard.tsx

krushna-simform replied:
I have renamed `DetailsPageMovieCard.tsx` to `MovieDetailsCard.tsx` (https://github.com/krushna-simform/filmfusion/pull/3/commits/94c9e2054b5f9dd49db747f0679dcc0a5673e4fd)

Kaushal-SF commented:
@krushna-simform Do we require this `Search Input` in the details page or in the `ticket booking` page? 

<img width="470" alt="image" src="https://github.com/user-attachments/assets/19c15bf1-ecca-49bb-80ac-b37f50b4efbb" />

krushna-simform commented:
> @krushna-simform Do we require this `Search Input` in the details page or in the `ticket booking` page?
> 
> <img alt="image" width="470" src="https://private-user-images.githubusercontent.com/97082607/442588091-19c15bf1-ecca-49bb-80ac-b37f50b4efbb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwMjU4NjYsIm5iZiI6MTc0NzAyNTU2NiwicGF0aCI6Ii85NzA4MjYwNy80NDI1ODgwOTEtMTljMTViZjEtZWNjYS00OWJiLTgwYWMtYjM3ZjUwYjRlZmJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEyVDA0NTI0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlYTdjZDU4MGI0Nzk1ZDMyOGZjMDdmN2ZhN2U2YmZlNmNmYzY1YzhlNzk0NjZiNTdkMDcyYmQyZmI3ZDcxYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BoETOq9pXC-1fJjKecvI1_9hV3vpVhCUXsaP47jjO9c">

I have taken reference from BookMyShow, and since they have a search bar on every page, I have included one on every page as well.

Kaushal-SF commented:
We can properly align the items in the details page. You can check the original website for inspiration.

<img width="461" alt="image" src="https://github.com/user-attachments/assets/2c331279-5238-468a-9b74-a9decf42baa7" />

Kaushal-SF commented:
> I have taken reference from BookMyShow, and since they have a search bar on every page, I have included one on every page as well.

Based on our requirement, do we need this search bar on every page? If no, we can omit it.

Kaushal-SF commented:
While creating a `404 not found` page, it is always a good idea to provide one button that help user to redirect to the `Home` page. I know we have a link in the Logo. However, user might not be aware about it.

<img width="300" alt="image" src="https://github.com/user-attachments/assets/4bc8240d-41db-41d6-96a6-9f4b806c75b2" />

Kaushal-SF commented:
Noticed one thing. What is the purpose of this numbers? Are they denoting likes or reviews or something else?

<img width="300" alt="image" src="https://github.com/user-attachments/assets/140dfbdf-d1dd-4604-ba86-ddae039090df" />

krushna-simform commented:
> Noticed one thing. What is the purpose of this numbers? Are they denoting likes or reviews or something else?
> 
> <img alt="image" width="300" src="https://private-user-images.githubusercontent.com/97082607/442592052-140dfbdf-d1dd-4604-ba86-ddae039090df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwMjY5MTEsIm5iZiI6MTc0NzAyNjYxMSwicGF0aCI6Ii85NzA4MjYwNy80NDI1OTIwNTItMTQwZGZiZGYtZDFkZC00NjA0LWJhODYtZGRhZTAzOTA5MGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEyVDA1MTAxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhMmM0ZDI4ODI1YjQzYTE4MjU2NmI3NDlkMjUxMThjNDE0M2ZmMTg5YTc2MDllM2VhYTkzNWY2MzNlMWMyMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JXfR-q47fTteNiklykUSYXHPTt0eGTqi3-D1JA52EyQ">

It is to show the total votes, and I will add a proper vote label after that.

krushna-simform committed:
FEATURE: add proper alignment to movieDetails page showcase section

krushna-simform committed:
FEATURE: add votes label in Rating component

krushna-simform committed:
FEATURE: add Search bar only inside mainFeed

krushna-simform commented:
> > I have taken reference from BookMyShow, and since they have a search bar on every page, I have included one on every page as well.
> 
> Based on our requirement, do we need this search bar on every page? If no, we can omit it.

I have added `SearchBar` only inside `MainFeed` (https://github.com/krushna-simform/filmfusion/pull/3/commits/a51639b6f063878321dbd30751a3e5f817314df9)

krushna-simform commented:
> We can properly align the items in the details page. You can check the original website for inspiration.
> 
> <img alt="image" width="461" src="https://private-user-images.githubusercontent.com/97082607/442588998-2c331279-5238-468a-9b74-a9decf42baa7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwMjc3NDYsIm5iZiI6MTc0NzAyNzQ0NiwicGF0aCI6Ii85NzA4MjYwNy80NDI1ODg5OTgtMmMzMzEyNzktNTIzOC00NjhhLTliNzQtYTlkZWNmNDJiYWE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEyVDA1MjQwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1Y2Y2MjI5MGYzNWRjMjkyMTI1NDNmZWZmZjU0YTYxZThiZWQ1Y2IzODA5MjY5MmJhYjU0YzdiNTUwOGE3YTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4XOX04HhHI849m9-hXlURT7LuVvSQnMSLtWyuisD1fM">

I have Aligned showcase section properly (https://github.com/krushna-simform/filmfusion/pull/3/commits/bed0d124f55893c4dfce0bace563692a36864de6)

krushna-simform commented:
> Noticed one thing. What is the purpose of this numbers? Are they denoting likes or reviews or something else?
> 
> <img alt="image" width="300" src="https://private-user-images.githubusercontent.com/97082607/442592052-140dfbdf-d1dd-4604-ba86-ddae039090df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwMjc3NDYsIm5iZiI6MTc0NzAyNzQ0NiwicGF0aCI6Ii85NzA4MjYwNy80NDI1OTIwNTItMTQwZGZiZGYtZDFkZC00NjA0LWJhODYtZGRhZTAzOTA5MGRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEyVDA1MjQwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI2ODE0YTc0ZDliYzM1ODE4ODBiMTVhMDViY2I5NDE0MWM4NmQ5NTMyYTBlMzliNTI5ZTJkNGU2MzY1ZWE1ZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AwmyVeTNgU3QyRgbROQXyHk5D19mEDpXXxuO8Rqx518">

I have added proper `Vote` label (https://github.com/krushna-simform/filmfusion/pull/3/commits/d01017317979c61d637ba19b273fa757a41a2b68)

krushna-simform committed:
FEATURE: add Back to Home Page button into 404 page not found component

krushna-simform commented:
> While creating a `404 not found` page, it is always a good idea to provide one button that help user to redirect to the `Home` page. I know we have a link in the Logo. However, user might not be aware about it.
> 
> <img alt="image" width="300" src="https://private-user-images.githubusercontent.com/97082607/442590648-4bc8240d-41db-41d6-96a6-9f4b806c75b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcwMjc3NDYsIm5iZiI6MTc0NzAyNzQ0NiwicGF0aCI6Ii85NzA4MjYwNy80NDI1OTA2NDgtNGJjODI0MGQtNDFkYi00MWQ2LTk2YTYtOWY0YjgwNmM3NWIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTEyVDA1MjQwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3OGIzMjc3NmM4NjUzYWQzZGY0MWQ3ZjAzOTA1MDJiZGE4OTg3ZmFkMWUzYmE1YmYwZDVlZmI4MmNkMjZhNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Xx5mrDx56dItQMmBIZJPgVP3rQPP_o0PDeylxgqI2nY">

I have added back to home page button (https://github.com/krushna-simform/filmfusion/pull/3/commits/911fe1672a641d3e0f3fa2bbbfa4014e7f462355)

Kaushal-SF commented on src/components/Header.tsx:
```diff
@@ -1,21 +1,22 @@
+import { Link, useLocation } from "react-router";
 import { Button } from "./ui/Button.tsx";
 import { Search } from "./ui/Search.tsx";
 
```
Nit: We don't need to add file extensions.

```suggestion
import { Button } from "./ui/Button";
import { Search } from "./ui/Search";
```

krushna-simform replied:
I have updated imports for `Button` and `Search` component (https://github.com/krushna-simform/filmfusion/pull/3/commits/504debe0ef184a81c7551dc54dfc76adf2ce7b5d)

Kaushal-SF commented on src/components/Header.tsx:
```diff
@@ -1,21 +1,22 @@
-        <Button ariaLabel="Login" innerText="Login" />
-        <Button ariaLabel="Register" innerText="Register" />
+        {location.pathname === "/" && <Search />}
+        <Button ariaLabel="Login" children="Login" />
```
Any specific reason why we are using the `children="Login` here?

krushna-simform replied:
My button component takes two props: one for the `ariaLabel` and another for the `children`.

![image](https://github.com/user-attachments/assets/cbac1287-f1c1-4a0d-9b0a-c53c5e9c207a)

Kaushal-SF replied:
@krushna-simform If this is just a `label` (name) for the button,  should we really need to pass it as a children?

krushna-simform replied:
Initially, when I was creating the button component, I thought it would be good to have children as a prop name, But i will update this prop name.

Kaushal-SF commented on src/components/MovieBooking.tsx:
```diff
@@ -0,0 +1,21 @@
+export const MovieBook = ({
+  title,
+  genre,
+}: Pick<MovieDetails, "title" | "genre">) => {
```
Same comment to take the type outside.

krushna-simform replied:
I have moved props type outside of the component (https://github.com/krushna-simform/filmfusion/pull/3/commits/714022c967c740329157f71fed2265a086e1902e)

Kaushal-SF commented on src/components/MovieCard.tsx:
```diff
@@ -1,55 +1,46 @@
-  const formattedTotalVotes = formatVotes(totalVotes);
-  const averageRating = ((positiveVotes / totalVotes) * 10).toFixed(1);
+  votes,
+}: Omit<MovieDetails, "cast" | "director" | "description">) => {
```
Same comment for this as well.

krushna-simform replied:
I have moved props types outside of the component (https://github.com/krushna-simform/filmfusion/pull/3/commits/34fb40c6ea7cb567c58b73cd43196a65351ef309)

Kaushal-SF commented on src/components/MovieDetailsDescription.tsx:
Can we called it as `MovieDescription`?

krushna-simform replied:
I have renamed `MovieDetailsDescription.tsx` to `MovieDescripition.tsx` (https://github.com/krushna-simform/filmfusion/pull/3/commits/038c980b1dbb595dd8e83b346f1bd9511e7818b1)

Kaushal-SF commented on src/components/MovieDetailsDescription.tsx:
```diff
@@ -0,0 +1,18 @@
+  description,
+  cast,
+  director,
+}: Pick<MovieDetails, "description" | "cast" | "director">) => {
```
Same comments as above. Please check for the other files.

krushna-simform replied:
I have moved props types outside of the component (https://github.com/krushna-simform/filmfusion/pull/3/commits/159f348bfad8d93c03eec304f9eb83a2e3231a8b)

Kaushal-SF commented on src/components/MovieDetailsShowcase.tsx:
```diff
@@ -0,0 +1,52 @@
+import { DetailsPageMovieCard } from "./DetailsPageMovieCard";
+import { Rating } from "./ui/Rating";
+import { Button } from "./ui/Button";
+import { useNavigate } from "react-router";
```
We can move this import to the top.

krushna-simform replied:
I have moved `{useNavigate}` import to the top (https://github.com/krushna-simform/filmfusion/pull/3/commits/bf6d50eb2368ea1a69e9aad53a9acaca129ea56c)

Kaushal-SF commented on src/components/ui/Image.tsx:
```diff
@@ -0,0 +1,15 @@
+interface ImageType {
```
```suggestion
interface ImageProps {
```

krushna-simform replied:
I have renamed interface `ImageType` to `ImageProps` (https://github.com/krushna-simform/filmfusion/pull/3/commits/ca9195968a2882b4e584ea342eedb7679e907d12)

Kaushal-SF commented on src/components/ui/MovieBookingSeats.tsx:
Perhaps, `MovieSeats.tsx`?

krushna-simform replied:
I have Renamed `MovieBookingSeats.tsx` to `MovieSeats.tsx` (https://github.com/krushna-simform/filmfusion/pull/3/commits/c137953a899aa36b21811405494c80622e30f854)

Kaushal-SF commented on src/components/ui/PageNotFound.tsx:
```diff
@@ -0,0 +1,14 @@
+      <p className="text-white text-9xl">404</p>
+      <p className="text-white text-2xl">Page not found</p>
+      <Link to="/" className="mt-6">
+        <Button ariaLabel="Back to Home page" children="Back to Home page" />
```
Same comment for usage of `children`. Any reason?

krushna-simform replied:
My button component takes two props: one for the `ariaLabel` and another for the `children`.

![image](https://github.com/user-attachments/assets/cbac1287-f1c1-4a0d-9b0a-c53c5e9c207a)

Kaushal-SF commented on src/data/movieData.ts:
Can provide more explicit name to the folder: `mockData`

krushna-simform replied:
I have renamed folder `src/data` to `src/mockData` (https://github.com/krushna-simform/filmfusion/pull/3/commits/885ccc62a04306ffaa0cf6126f2d7038e33a4c8d)

Kaushal-SF commented on src/pages/MainFeedPage/MainFeedPage.tsx:
No need to mention the Page at the last. We can simply say `MainFeed` only.
Same for the other files as well.

krushna-simform replied:
I have renamed 
- `MainFeedPage.tsx` to `MainFeed.tsx` (https://github.com/krushna-simform/filmfusion/pull/3/commits/6b59c86e3dd03d489d31dea632bc2d969a3ad0c7)
- `MovieBookingPage.tsx` to `MovieBooking.tsx` (https://github.com/krushna-simform/filmfusion/pull/3/commits/09af5f5d40ed09c375b66e168b69de43af37e2c9)
- `MovieDetailsPage.tsx` to `MovieDetils.tsx` (https://github.com/krushna-simform/filmfusion/pull/3/commits/e39c9d70a58a86e10fd4174ad213740761c8b7cd)

Kaushal-SF commented on src/utils/interface.ts:
Should we put types file in `utils` folder?

krushna-simform replied:
I have renamed  folder `utils` to `types`(https://github.com/krushna-simform/filmfusion/pull/3/commits/a0acbe8ecdb77e397da86bfcf91b32df41e93cb4)

Kaushal-SF reviewed (commented):
@krushna-simform Overall good work!

UI is looking wonderful. Good work with the component designing.
However, I found few ares where code can be improved.

krushna-simform committed:
FEATURE: rename utils folder to types

krushna-simform committed:
FEATURE: move card props type outside of the component

krushna-simform committed:
FEATURE: update imports for button and search component

krushna-simform committed:
FEATURE: move MovieBooks props type outside of the component

krushna-simform committed:
FEATURE: move MovieCard props type outside of the component

krushna-simform committed:
FEATURE: move MovieDescriptionProps type outside of the component

krushna-simform committed:
FEATURE: movie useNavigate import to the top

krushna-simform committed:
FEATURE: update Image component interface name

krushna-simform committed:
FEATURE: rename DetailsPageMovieCard.tsx to MovieDetailsCard.tsx and update MovieDetailsShowcase based on reaname

krushna-simform committed:
FEATURE: rename MovieDetailsDescription to MovieDescription, and update MovieDetailsPage based on rename

krushna-simform committed:
FEATURE: rename MovieBookingSeats to MovieSeats, and update MovieBooking.tsx based on rename

krushna-simform committed:
FEATURE: rename folder src/data to src/mockData, and update other components based on rename

krushna-simform committed:
FEATURE: rename MainFeedPage.tsx to MainFeed.tsx, and update main.tsx based on rename

krushna-simform committed:
FEATURE: rename MovieBookingPage.tsx to MovieBooking.tsx, and update main.tsx based on rename

krushna-simform committed:
FEATURE: rename MovieDetailsPage.tsx to MovieDetails.tsx, and update main.tsx based on rename

Kaushal-SF reviewed (commented):
@krushna-simform Changes look good. Added one last comment.

Kaushal-SF reviewed (approved):
Overall good work. Approving the PR!
